arangod/Network/Methods.cpp

```diff
@@ -134,8 +134,10 @@ FutureRes sendRequest(ConnectionPool* pool, DestinationId dest, RestVerb type,
 
     arangodb::network::EndpointSpec spec;
     int res = resolveDestination(*pool->config().clusterInfo, dest, spec);
-    if (res != TRI_ERROR_NO_ERROR) {  // FIXME return an error  ?!
-      return futures::makeFuture(Response{std::move(dest), Error::Canceled, nullptr, std::move(req)});
+    if (res != TRI_ERROR_NO_ERROR) {
+      auto errorPromise = futures::Promise<network::Response>::makeEmpty();
+      errorPromise.setException(arangodb::basics::Exception(res, __FILE__, __LINE__));
+      return errorPromise.getFuture();
```

Comment:
```suggestion
      return makeFuture<network::Response>(arangodb::basics::Exception(res, __FILE__, __LINE__));
```

Reply:
That is better, thank you, included the fix, as we need to remove the lines above in addition. :+1: